File: Modulo15/Vector-Pessoa/Coordenador.cpp
#include "Coordenador.h"
#include <iostream>
using namespace std;

Coordenador::Coordenador(string nome, int CPF, double salario, string curso) : 
    Professor(nome, CPF, salario), curso(curso) {
}

Coordenador::~Coordenador() {
    cout << "Destrutor Coordenador: (" << curso << ")" << endl;
}

double Coordenador::getSalario() const {
    return Professor::getSalario() + 2000;
}

string Coordenador::getCurso() const {
    return curso;
}

void Coordenador::imprime() const {
    Professor::imprime();
    cout << "Curso: " << getCurso() << endl;
}

File: Modulo12/Professor-CSharp/Coordenador.cs
using System;

namespace poo {

    class Coordenador : Professor {

        private string curso;
        public Coordenador(string nome, int CPF, double salario, string curso) : base(nome, CPF, salario) {
            this.curso = curso;
        }
        public override double getSalario() {
            return base.getSalario() + 2000;
        }
        public string getCurso() {
            return curso;
        }

        public override void imprime() {
            base.imprime();
            Console.WriteLine("Curso: " + curso);
        }
    }

}